journal.py

```diff
@@ -86,22 +94,41 @@ def home(request):
 
 @view_config(route_name='add', renderer='templates/add.jinja2')
 def add(request):
-    entries = Entry.all()
-    return {'entries': entries}
+    if request.method == 'POST':
+        title = request.params.get('title')
+        text = request.params.get('text')
+        if not (title == "" or text == ""):
+            Entry.write(title=title, text=text)
+            return HTTPFound(request.route_url('home'))
+        else:
+            return {'title': title, 'text': text}
```

Comment:
good! you could also return errors, to explain why the user wasn't redirected as they expected.